### PR TITLE
export code reference classes at dagster package

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Adds top-level exports for various code reference related classes and functions.

### CODE REVIEW

**Comment by benpankow** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21881?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#21881** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21881?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#21713** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21713?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21675** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21675?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21468** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21468?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21467** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21467?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21465** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21465?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21576** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/21576?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by schrockn** [comment]
Can we get some reps (meaning actual use) on these APIs before exporting them top-level?

**Comment by sryza** [comment]
@schrockn do you mean internally within Dagster Labs? I generally don't think we should be pointing users at internal APIs.

These will be experimental, so it doesn't seem very high risk.

**Comment by schrockn** [comment]
I've soured a bit on relying "experimental" so much as it pollutes the global namespace. If we have high touch users that we are in close contact with I think pointing them at internal APIs is ok. Let's follow up in slack/email.

File: python_modules/dagster/dagster/__init__.py [hunk 1]
```diff
@@ -289,7 +293,10 @@
     TableSchemaMetadataValue as TableSchemaMetadataValue,
     TextMetadataValue as TextMetadataValue,
     TimestampMetadataValue as TimestampMetadataValue,
+    UrlCodeReference as UrlCodeReference,
     UrlMetadataValue as UrlMetadataValue,
+    link_to_git as link_to_git,
```
[the remark this answers is not in the document]
Comment (benpankow) [reply python_modules/dagster/dagster/__init__.py]: Updated

**Pull request merged**